data/modules/SoldOut/SoldOut.lua

```diff
@@ -89,9 +94,13 @@ local makeAdvert = function(station, commodity)
 
 	-- Amount that the desperado wants to buy is determined by the
 	-- wallet, i.e. fewer units if expensive commodity.
-	-- Money is uniformly distirubted in log space
-	local money_to_spend = lograndom(1e2, 1e5)
-	ad.amount = math.ceil(money_to_spend / math.abs(ad.price))
+	-- Amount should not exceed what station normally wants (relevant values at line 57 (former line 48 of SpaceStation.lua).
+	-- Actually we set to desperately want up to what a Major exports offer at double its price. ad.price/2 because it is already doubled.
+	-- Replaces the use of lograndom function.
+	local maxMoney = math.max(math.abs(ad.price)+1.00, Calibration.InvestmentOfPrice(math.abs(ad.price)/2) * avgStock * 2)
+	local money_to_spend = Engine.rand:Number(math.abs(ad.price), maxMoney)
+
+	ad.amount = math.min(math.ceil(money_to_spend / math.abs(ad.price)), Game.player.totalCargo * 2)
```

Comment:
Now this I like, better metrics for cargo missions are always welcome, as rewards for in-system hauling, etc. are a bit fudged.

Reply:
You are aware of course that exponential is used to calculate InvestmentOfPrice.

Writing this I am pretty sure that If in Calibration.lua we set
```
local Rab = 1
StockPriceOne = 10^5
```
let missions' rewards untouched, and prerequisite table empty (deleting lines 73-87 of it) we should have current state untouched prepared for altering these values when game is ready for them.